### DISCUSSION TITLE
Render buffer as image

### DISCUSSION THREAD

**Discussion by guerinoni** [emilk/egui#679] [Q&A] [answered]
I'm writing an emulator and I need to render on-screen a buffer like Vec<u8>... How I can do this? Using egui::Image? Or Texture? I'm reading all docs but I don't have found anything related.
Thanks in advance

**Comment by emilk** [comment, marked as answer]
Are you using `eframe`? If so, use the texture allocator: https://docs.rs/epi/0.14.0/epi/struct.Frame.html#method.tex_allocator together with `egui::Image`

**Reply by guerinoni** [reply]
Can you provide a simple example?? 
If I have something like
```
const WIDTH: usize = 640;
const HEIGHT: usize = 360;
let mut buffer: Vec<u32> = vec![0; WIDTH * HEIGHT];
```

Now I'm searching for a sort of `window.update(buffer)`

**Reply by emilk** [reply]
Here is an example of loading an image: https://github.com/emilk/egui/blob/master/examples/retained_image/src/main.rs

**Reply by guerinoni** [reply]
Many thanks, dude! Very helpful!! :)

**Reply by umberto-fontanazza** [reply]
Is this reply outdated?

**Reply by flameoftheforest** [reply]
this seems outdate:
https://github.com/emilk/egui/blob/master/examples/retained_image/src/main.rs

not sure if this helps:
https://github.com/emilk/egui/tree/master/examples/screenshot
